semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -113,6 +115,7 @@ let generic_to_json env (key : key) ast =
                  match x.G.e with
                  | G.Container
                      (G.Tuple, (_, [ { e = L (String (k, _)); _ }; v ], _)) ->
+                     (* BUG: need to unescape the string k *)
```

Comment:
Actually we don't need to do it when the generic AST comes from Yaml_to_generic.